students/chris_lombardi/Lesson09/donor_models.py

```diff
@@ -0,0 +1,103 @@
+class Donor:
+    """
+    A class for a donor object
+    Stores the name of the donor and donation history data.
+    """
+
+    def __init__(self, full_name):
+        """
+        Creates a donor.
+
+        Keyword argument:
+        full_name -- First and last name of a donor.
+        """
+
+        self.name = full_name.title()
+        self.donation_hist = []
+
+    def name(self):
+        """Return full name of donor."""
+        return self.name
+
+    def donation_hist(self):
+        """Return a list of all donations."""
+        return self.donation_hist
+
+    @property
+    def total_donation(self):
+        """A sum of all donations made by the donor."""
+        return sum(self.donation_hist)
+
+    @property
+    def num_donation(self):
+        """Return a count of individual donations made."""
+        return len(self.donation_hist)
+
+    @property
+    def ave_donation(self):
+        """Return the average donation for the donor."""
+        #Checks that there is at least one donation for an average.
+        if self.num_donation == 0:
+            return 0
+        else:
+            return (self.total_donation/self.num_donation)
+
+    def add_donation(self, *args):
+        """Add a variable number of donations to the donation history."""
+        self.donation_hist.extend(args)
+
+    def __lt__(self, other):
+        return self.total_donation < other.total_donation
+
+    def __repr__(self):
+        return self.name
+
+    def send_thank_you(self, don_amnt):
+        """Sends a thank you note when a donation is made."""
+        ty_str = []
+        ty_str.append(f'\nDear {self.name.title()},\n')
+        ty_str.append(f'Thank you for your generous donation ' +
+          'of ${don_amnt:,.2f}!\nWe appreciate your contribution to our ' +
+          'charity.\n\nSincerley,\nThe Mailroom\n')
+        return ''.join(ty_str)
+
+
+class DonorCollection:
+
+    """
+    A class containing information of a list of Donor objects
+    """
+
+    def __init__(self):
+        """Create a collection of donor objects."""
+        self.list_donors = []
+
+    def list_donors(self):
+        """Return a list of all donors in the collection."""
+        return list_donors
```

Comment:
looks like this could be best as a `property`? otherwise both your variable and method names are the same name which is confusing.